lib/Runtime/Library/InJavascript/Intl.js

```diff
@@ -1593,44 +1593,122 @@
                 dateTimeFormat.__initializedDateTimeFormat = true;
             }
 
-            platform.registerBuiltInFunction(tagPublicFunction("Date.prototype.toLocaleString", function () {
-                if (typeof this !== 'object' || !(this instanceof Date)) {
-                    platform.raiseNeedObjectOfType("Date.prototype.toLocaleString", "Date");
-                }
-                var value = callInstanceFunc(DateInstanceGetDate, new Date(this));
-                if (isNaN(value) || !isFinite(value)) {
-                    return "Invalid Date";
-                }
-                var stateObject = setPrototype({}, null);
-                InitializeDateTimeFormat(stateObject, arguments[0], ToDateTimeOptions(arguments[1], "any", "all"));
-                return String(platform.formatDateTime(Internal.ToNumber(this), stateObject));
-            }), 0);
+            {
+                // scope for the cache variable
+                let __date_prototype_toLocaleString_default_formatterStateObject_cache = undefined;
+                platform.registerBuiltInFunction(tagPublicFunction("Date.prototype.toLocaleString", function () {
+                    if (typeof this !== 'object' || !(this instanceof Date)) {
+                        platform.raiseNeedObjectOfType("Date.prototype.toLocaleString", "Date");
+                    }
+                    var value = callInstanceFunc(DateInstanceGetDate, new Date(this));
```

Comment:
nit: if we are adding new code, we should probably be using let/const consistently

Reply:
Yes, I meant to update these. Thanks.